ingestion-edge/pubsub_emulator.py

```diff
@@ -28,14 +28,33 @@ class PubsubEmulator(
 ):
     """Pubsub gRPC emulator for testing."""
 
-    def __init__(self, server: grpc.Server):
+    def __init__(
+        self,
+        max_workers: int = int(os.environ.get("MAX_WORKERS", 1)),
+        port: int = int(os.environ.get("PORT", 0)),
+    ):
         """Initialize a new PubsubEmulator and add it to a gRPC server."""
         self.topics: Dict[str, Set[Subscription]] = {}
         self.subscriptions: Dict[str, Subscription] = {}
         self.status_codes: Dict[str, grpc.StatusCode] = {}
         self.sleep = None
-        pubsub_pb2_grpc.add_PublisherServicer_to_server(self, server)
-        pubsub_pb2_grpc.add_SubscriberServicer_to_server(self, server)
+        self.port = port
+        self.max_workers = max_workers
+        self.new_server()
+
+    def new_server(self):
```

Comment:
This name being a noun sounds like it should return some object representing the server. I'd prefer this be a verb like `create_server` or `initialize_server`.

Reply:
And I now see that this is replacing a `create_server` that _did_ return a value 😆

Reply:
I'll rename it back to create_server